tests_new/integration_tests/core/environment/test_environment.py

```diff
@@ -51,15 +54,18 @@ def test_list_envs_invited(client2, session_env1, session_env2, session_id):
 
 
 def test_persistent_env_update(client1, persistent_env1):
-    # wait for stack to get to a final state before triggering an update
-    stack_uri = persistent_env1.stack.stackUri
-    env_uri = persistent_env1.environmentUri
-    check_stack_ready(client1, env_uri, stack_uri)
-    update_stack(client1, env_uri, 'environment')
-    # wait for stack to move to "in_progress" state
-    check_stack_in_progress(client1, env_uri, stack_uri)
-    stack = check_stack_ready(client1, env_uri, stack_uri)
-    assert_that(stack.status).is_equal_to('UPDATE_COMPLETE')
+    stack = get_stack(
+        client1,
+        persistent_env1.environmentUri,
+        persistent_env1.stack.stackUri,
+        persistent_env1.environmentUri,
+        target_type='environment',
+    )
+    updated_before = datetime.strptime(stack.updated, '%Y-%m-%d %H:%M:%S.%f').timestamp()
+    stack = update_env_stack(client1, persistent_env1)
+    assert_that(stack).contains_entry(status='UPDATE_COMPLETE')
+    updated = datetime.strptime(stack.updated, '%Y-%m-%d %H:%M:%S.%f').timestamp()
+    assert_that(updated).is_greater_than_or_equal_to(updated_before)
```

Comment:
1. we write ISO dates so you can use `datetime.fromisoformat(...)`
2. you don't have to convert to timestamps, datetime objects are comparable and errors will much easier to read 
3. depending on whether `updated` is updated based on CFN or not (see my comment below) we might not want the "equal" part

i.e...
```python
nn1 = datetime.datetime.fromisoformat('2024-10-01 08:12:27.732312') # this is the format we write in the db
nn2 = datetime.datetime.now()
assert_that(nn2).is_greater_than(nn1)
```

Reply:
1. ok
2. ok
3. We need equal)

Reply:
wrt `3` and for the record (not to be solved in this PR).

Ideally we would want to be able to tell wether an update was issued but at the moment I don't think it's possible as this is a limitation of the backend.